src/nbl/video/COpenGLSync.h

```diff
@@ -39,7 +39,7 @@ class COpenGLSync final : public core::IReferenceCounted
             if (cachedRetval != ES_TIMEOUT_EXPIRED)
                 return cachedRetval;
 
-            GLenum status = _gl->glSync.pglClientWaitSync(sync, 0, timeout);
+            GLenum status = _gl->glSync.pglClientWaitSync(sync, 0, timeout); // GL_SYNC_FLUSH_COMMANDS_BIT to flags?
```

Comment:
the OpenGL spec actually says that this flush is only peformed when you use the sync like you would use an event (only works if its within the same context) so you should only set the flush bit the first time you wait from an IGPUEvent

Reply:
but in the case of an event you'd want to remove the call to `_gl->glGeneral.pglFlush();`, ensure the context which sets the event actually creates the `COpenGLSync`  and make the constructor something like `COpenGLSync(IOpenGL_LogicalDevice* _dev, IOpenGL_FunctionTable* _gl, will_only_use_for_an_event_t will_only_use_for_an_event)` and set some sort of a boolean/record the queue (or the function table) to ensure you can check that any cpu or gpu wait is done with the same queue ( and crash/throw a big fat error if it isnt), the GL_SYNC_FLUSH_COMMANDS_BIT  can be set in the clientwaitsync, and turn the GPU wait into a NOOP.